numpy/core/_add_newdocs.py

```diff
@@ -5491,6 +5491,39 @@
 
     """))
 
+add_newdoc('numpy.core.multiarray', 'dtype', ('metadata',
+    """
+    Either ``None`` or a readonly dictionary of metadata (mappingproxy).
+
+    The metadata field can be set using any dictionary at data-type
+    creation. Note that whether or not operations on arrays with metadata
+    attached to their datatypes is currently not well defined and should
+    not be relied on.
```

Comment:
Alternative?
```suggestion
    creation. NumPy currently has no uniform approach to metadata; although
    some array operations preserve it there is no guarantee that others will. 
```

Reply:
This is a stronger guarantee than what I suggest in https://github.com/numpy/numpy/pull/17182#discussion_r479645917 - I think @seberg's point was not just "it sometimes is and sometimes isn't" but also "the times when it is and isn't may change in future without warning".

Reply:
> the times when it is and isn't may change in future without warning

In other words, it's unusable. 
* If so, what is the motive in documenting it?
* Why bother with examples if we intend for no one to use it?

> I think @seberg's point

You or I or both of us may be mistaken. Perhaps we can leave a little oxygen in the room for @seberg .

Reply:
Some people use it (h5py), I have no idea if people use it successfully. Right now there are some things which will have to change.
I am not too worried about the shift in semantics, but propagating metadata to the output seems fraught with issues right now (which unfortunately doesn't guarantee nobody is doing it).

Reply:
So this is literally an undocumented feature that's been picked up and used in a project?

What's our goal in writing about it now? "metadata was an undocumented dtype option that was discovered and used 
in outside projects. It is poorly supported and aspects of its behavior are slated to change."  That explains a lot to anyone who sees it h5py or in the docs.  Since we're good people we can try to state what parts we expect to stay working, what will change (when and how if possible), and declare everything else to be permanently at risk and not our problem.

Reply:
Well, I though I could do it, because I will change some details right now, and its weird not to documented around for a decade.

I adopted your suggestions (with some modification), also for the first introduction.  And the quote from above as a "warning" box.  That is a bit strong probably, but maybe actually better than the alternative of just not documenting it...